File: C#/OpenApi.Utility/PropertyFormater.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace OpenApi.Utility
{
    /// <summary>
    /// 属性格式化
    /// </summary>
    public static class PropertyFormater
    {
        /// <summary>
        /// 对象格式化
        /// </summary>
        /// <param name="name">属性名称</param>
        /// <param name="value">属性值</param>
        /// <param name="valueType">属性类型</param>
        /// <returns></returns>
        public static string ObjectFormat(string name, object value, ValueTypeEnum valueType)
        {
            object formatValue = null;
            switch (valueType)
            {
                case ValueTypeEnum.DateTime: formatValue = DateTimeFormat(value); break;
                case ValueTypeEnum.Single: formatValue = FloatFormat(value); break;
                case ValueTypeEnum.Double: formatValue = DoubleFormat(value); break;
                case ValueTypeEnum.Decimal: formatValue = DecimalFormat(value); break;
                case ValueTypeEnum.Boolean: formatValue = BooleanFormat(value); break;
                case ValueTypeEnum.Guid: formatValue = GuidFormat(value); break;
                case ValueTypeEnum.SByte:
                case ValueTypeEnum.Int16:
                case ValueTypeEnum.Int32:
                case ValueTypeEnum.Int64:
                case ValueTypeEnum.Byte:
                case ValueTypeEnum.UInt16:
                case ValueTypeEnum.UInt32:
                case ValueTypeEnum.UInt64:
                case ValueTypeEnum.Char:
                case ValueTypeEnum.String:
                    formatValue = value; break;
                default: break;
            }

            return formatValue == null ? string.Empty : string.Format("{0}{1}", name, formatValue);
        }

        #region 各类型格式化方法

        /// <summary>
        /// DateTime格式化
        /// </summary>
        /// <param name="obj"></param>
        /// <returns></returns>
        public static int DateTimeFormat(object obj)
        {
            DateTime datetime;
            if (obj is DateTime)
                datetime = (DateTime)obj;
            else
                datetime = Convert.ToDateTime(obj);

            TimeSpan span = datetime - new DateTime(1970, 1, 1);
            int timestamp = Convert.ToInt32(span.TotalSeconds);

            return timestamp;
        }

        /// <summary>
        /// Float格式化
        /// </summary>
        /// <param name="obj"></param>
        /// <returns></returns>
        public static string FloatFormat(Object obj)
        {
            Single real;
            if (obj is Single)
                real = (Single)obj;
            else
                real = Convert.ToSingle(obj);
            
            byte[] byteArray = BitConverter.GetBytes(real);
            
            if (BitConverter.IsLittleEndian)
                Array.Reverse(byteArray);

            return ByteArrayToHexString(byteArray);
        }

        /// <summary>
        /// Double格式化
        /// </summary>
        /// <param name="obj"></param>
        /// <returns></returns>
        public static string DoubleFormat(Object obj)
        {
            Double real;
            if (obj is Double)
                real = (Double)obj;
            else
                real = Convert.ToDouble(obj);

            byte[] byteArray = BitConverter.GetBytes(real);

            if (BitConverter.IsLittleEndian)
                Array.Reverse(byteArray);

            return ByteArrayToHexString(byteArray);
        }

        /// <summary>
        /// Decima格式化
        /// </summary>
        /// <param name="obj"></param>
        /// <returns></returns>
        public static string DecimalFormat(Object obj)
        {
            Double real = Convert.ToDouble(obj);
            return DoubleFormat(real);
        }

        /// <summary>
        /// Boolean格式化
        /// </summary>
        /// <param name="obj"></param>
        /// <returns></returns>
        public static int BooleanFormat(Object obj)
        {
            Boolean real;
            if (obj is Boolean)
                real = (Boolean)obj;
            else
                real = Convert.ToBoolean(obj);

            return real ? 1 : 0;
        }

        /// <summary>
        /// Guid格式化
        /// </summary>
        /// <param name="obj"></param>
        /// <returns></returns>
        public static string GuidFormat(Object obj)
        {
            Guid real;
            if (obj is Guid)
                real = (Guid)obj;
            else
                real = new Guid(obj.ToString());

            return real.ToString("D");
        }

        private static string ByteArrayToHexString(byte[] ba)
        {
            StringBuilder hex = new StringBuilder(ba.Length * 2);
            foreach (byte b in ba)
                hex.AppendFormat("{0:x2}", b);
            return hex.ToString().ToUpper();
        }

        #endregion
    }
}

File: Java/src/main/java/com/ppdai/open/core/PropertyFormater.java
package com.ppdai.open.core;

import org.apache.commons.codec.binary.Hex;

import java.nio.ByteBuffer;
import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.UUID;

/**
 * 属性格式化类
 * Created by xuzhishen on 2016/5/10.
 */
public class PropertyFormater {

    /**
     * 对象格式化
     * @param name  属性名称
     * @param value 属性值
     * @param valueType 属性类型
     * @return  格式化后的字符创
     * @throws ParseException
     */
    public static String ObjectFormat(String name, Object value, ValueTypeEnum valueType) throws ParseException {
        Object formatValue = null;
        switch (valueType) {
            case DateTime:
                formatValue = dateTimeFormat(value);
                break;
            case Single:
                formatValue = floatFormat(value);
                break;
            case Double:
                formatValue = doubleFormat(value);
                break;
            case Decimal:
                formatValue = decimalFormat(value);
                break;
            case Boolean:
                formatValue = booleanFormat(value);
                break;
            case Guid:
                formatValue = guidFormat(value);
                break;
            case SByte:
            case Int16:
            case Int32:
            case Int64:
            case Byte:
            case UInt16:
            case UInt32:
            case UInt64:
            case Char:
            case String:
                formatValue = value;
                break;
            default:
                break;
        }

        return formatValue == null ? "" : String.format("%s%s", name, formatValue);
    }

    /**
     * 日期时间格式化
     * @param obj   待格式化对象
     * @return
     * @throws ParseException
     */
    public static long dateTimeFormat(Object obj) throws ParseException {
        SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd");
        Date real = null;
        if (obj instanceof Date)
            real = (Date) obj;
        else
            real = sdf.parse(obj.toString());

        return (real.getTime() - sdf.parse("1970-01-01").getTime())/1000;
    }

    /**
     * 浮点格式化
     * @param obj   待格式化对象
     * @return
     */
    public static String floatFormat(Object obj) {
        Float real = null;
        if (obj instanceof Float)
            real = (Float) obj;
        else
            real = Float.parseFloat(obj.toString());

        return Hex.encodeHexString(toByteArray(real)).toUpperCase();
    }

    /**
     * 双精度格式化
     * @param obj   待格式化对象
     * @return
     */
    public static String doubleFormat(Object obj) {
        Double real = null;
        if (obj instanceof Double)
            real = (Double) obj;
        else
            real = Double.parseDouble(obj.toString());

        return Hex.encodeHexString(toByteArray(real)).toUpperCase();
    }

    /**
     * 布尔格式化
     * @param obj   待格式化对象
     * @return
     */
    public static int booleanFormat(Object obj) {
        Boolean real = null;
        if (obj instanceof Boolean)
            real = (Boolean) obj;
        else
            real = Boolean.parseBoolean(obj.toString());

        return real ? 1 : 0;
    }

    /**
     * Decimal格式化
     * @param obj   待格式化对象
     * @return
     */
    public static String decimalFormat(Object obj) {
        return doubleFormat(obj);
    }

    /**
     * Guid格式化
     * @param obj   待格式化对象
     * @return
     */
    public static String guidFormat(Object obj) {
        UUID real = null;
        if (obj instanceof UUID)
            real = (UUID) obj;
        else
            real = UUID.fromString(obj.toString());

        return real.toString();
    }

    /**
     * 浮点类型转换成字节数组
     * @param val
     * @return
     */
    public static byte[] toByteArray(float val) {
        byte[] bytes = new byte[4];
        ByteBuffer.wrap(bytes).putFloat(val);
        return bytes;
    }

    /**
     * 双精度类型转换成字节数组
     * @param val
     * @return
     */
    public static byte[] toByteArray(double val) {
        byte[] bytes = new byte[8];
        ByteBuffer.wrap(bytes).putDouble(val);
        return bytes;
    }
}
